server/src/main/java/org/opentosca/toscana/plugins/cloudformation/CloudFormationLifecycle.java

```diff
@@ -10,59 +10,97 @@
+import org.opentosca.toscana.plugins.cloudformation.visitor.CheckModelNodeVisitor;
+import org.opentosca.toscana.plugins.cloudformation.visitor.CheckModelRelationshipVisitor;
 import org.opentosca.toscana.plugins.cloudformation.visitor.CloudFormationNodeVisitor;
+import org.opentosca.toscana.plugins.cloudformation.visitor.PrepareModelNodeVisitor;
+import org.opentosca.toscana.plugins.cloudformation.visitor.PrepareModelRelationshipVisitor;
 import org.opentosca.toscana.plugins.util.TransformationFailureException;
 
 import com.amazonaws.auth.AWSCredentials;
 import com.amazonaws.auth.BasicAWSCredentials;
+import org.jgrapht.Graph;
 
 import static org.opentosca.toscana.plugins.cloudformation.CloudFormationPlugin.AWS_ACCESS_KEY_ID_KEY;
 import static org.opentosca.toscana.plugins.cloudformation.CloudFormationPlugin.AWS_REGION_KEY;
 import static org.opentosca.toscana.plugins.cloudformation.CloudFormationPlugin.AWS_SECRET_KEY_KEY;
 
 public class CloudFormationLifecycle extends AbstractLifecycle {
     private final EffectiveModel model;
-    private String awsRegion;
-    private AWSCredentials awsCredentials;
+    private CloudFormationModule cfnModule;
+    private PluginFileAccess fileAccess;
 
     public CloudFormationLifecycle(TransformationContext context) throws IOException {
         super(context);
         model = context.getModel();
-        if (context.getProperties() == null) {
-            //lifecycle test failes because getProperties is null
-            awsRegion = "us-west-2";
-            awsCredentials = new BasicAWSCredentials("", "");
-            return;
-        }
         Map<String, String> properties = context.getProperties().getPropertyValues();
-        awsRegion = properties.get(AWS_REGION_KEY);
+        String awsRegion = properties.get(AWS_REGION_KEY);
         String keyId = properties.get(AWS_ACCESS_KEY_ID_KEY);
         String secretKey = properties.get(AWS_SECRET_KEY_KEY);
-        awsCredentials = new BasicAWSCredentials(keyId, secretKey);
+        AWSCredentials awsCredentials = new BasicAWSCredentials(keyId, secretKey);
+        this.fileAccess = context.getPluginFileAccess();
+        this.cfnModule = new CloudFormationModule(fileAccess, awsRegion, awsCredentials);
+    }
+
+    /**
+     Convert input to alphanumerical string
+     */
+    public static String toAlphanumerical(String inp) {
+        return inp.replaceAll("[^A-Za-z0-9]", "");
     }
 
     @Override
     public boolean checkModel() {
-//        TODO implement model checks
+        logger.info("Check model for compatibility to CloudFormation");
+        Set<RootNode> nodes = model.getNodes();
+        Set<RootRelationship> relationships = model.getTopology().edgeSet();
+        try {
+            CheckModelNodeVisitor checkModelNodeVisitor = new CheckModelNodeVisitor(context);
+            logger.debug("Check nodes");
+            for (VisitableNode node : nodes) {
+                node.accept(checkModelNodeVisitor);
```

Comment:
In the Visitor classes you always log which node you are visiting when entering the method. Why don't you just write it here once for all methods?